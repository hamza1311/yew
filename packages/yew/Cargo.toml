[package]
name = "yew"
version = "0.17.4"
edition = "2018"
authors = [
  "Denis Kolodin <deniskolodin@gmail.com>",
  "Justin Starry <justin@yew.rs>",
]
repository = "https://github.com/yewstack/yew"
homepage = "https://github.com/yewstack/yew"
documentation = "https://docs.rs/yew/"
license = "MIT/Apache-2.0"
readme = "../README.md"
keywords = ["web", "webasm", "javascript"]
categories = ["gui", "wasm", "web-programming"]
description = "A framework for making client-side single-page apps"

[dependencies]
anyhow = "1"
console_error_panic_hook = "0.1"
gloo = "0.2.1"
indexmap = { version = "1.5", features = ["std"] }
js-sys = "0.3"
log = "0.4"
wasm-bindgen = "0.2.60"
wasm-bindgen-futures = "0.4"
yew-macro = { version = "0.17.0", path = "../yew-macro" }

[dependencies.web-sys]
version = "0.3"
features = [
  "AbortController",
  "AbortSignal",
  "AnimationEvent",
  "BinaryType",
  "Blob",
  "BlobPropertyBag",
  "console",
  "DedicatedWorkerGlobalScope",
  "Document",
  "DomTokenList",
  "DragEvent",
  "Element",
  "ErrorEvent",
  "Event",
  "EventTarget",
  "File",
  "FileList",
  "FileReader",
  "FocusEvent",
  "Headers",
  "HtmlElement",
  "HtmlButtonElement",
  "HtmlInputElement",
  "HtmlSelectElement",
  "HtmlTextAreaElement",
  "InputEvent",
  "KeyboardEvent",
  "Location",
  "MessageEvent",
  "MouseEvent",
  "Node",
  "ObserverCallback",
  "PointerEvent",
  "ProgressEvent",
  "ReferrerPolicy",
  "Request",
  "RequestCache",
  "RequestCredentials",
  "RequestInit",
  "RequestMode",
  "RequestRedirect",
  "Response",
  "Storage",
  "Text",
  "TouchEvent",
  "TransitionEvent",
  "UiEvent",
  "Url",
  "WebSocket",
  "WheelEvent",
  "Window",
  "Worker",
  "WorkerGlobalScope",
  "WorkerOptions",
]

[dev-dependencies]
trybuild = "1.0"
wasm-bindgen-test = "0.3.4"

[features]
default = []
doc_test = []
wasm_test = []
wasm_bench = []

[package.metadata.docs.rs]
features = ["doc_test"]
