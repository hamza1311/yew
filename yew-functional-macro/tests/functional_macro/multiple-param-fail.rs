use yew::prelude::*;
use yew_functional::functional_component;

#[derive(Clone, Properties, PartialEq)]
struct Props {
    a: usize,
}

#[functional_component(Comp)]
fn comp(props: &Props, invalid: String) -> Html {
    html! {
        <p>
            { props.a }
            { invalid }
        </p>
    }
}

#[functional_component(Comp)]
fn comp3(props: &Props, invalid: String, another_invalid: u32) -> Html {
    html! {
        <p>
            { props.a }
            { invalid }
            { another_invalid }
        </p>
    }
}

fn main() {}
