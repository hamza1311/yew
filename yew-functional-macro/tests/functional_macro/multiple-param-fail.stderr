error: functional components can accept at most one parameter for the props
  --> $DIR/multiple-param-fail.rs:10:24
   |
10 | fn comp(props: &Props, invalid: String) -> Html {
   |                        ^^^^^^^^^^^^^^^
